### PR TITLE
Update tinyMCE to version 4.2.6

### PR DESCRIPTION
**Pull request by dgrammatiko**
#### Sync with the latest release

This is an update from 4.1.6 to the latest version: 4.2.6
There are a lot of bug fixes and also a new flat theme

Here is a preview:
screen shot 2015-10-05 at 09 11 45

@infograf768 This is a bug fix release so there shouldn’t be any new language strings here (that was already done in 4.2)
##### Testing

Please don’t use patch tester for this one. You need to use Git here to get the commit (or manually overwrite the files)

An even easier method: download https://github.com/dgt41/joomla-cms/archive/tinyMCE_4.2.3.zip and install as usual

### CODE REVIEW

**Comment by infograf768** [comment]
@dgt41 
Don't we still need the tiny lang updates before next release as we are still in core at version 4.1.7? This has to be organised by Mig.

**Comment by dgt41** [comment]
@infograf768 Oh sure, the stings introduced in 4.2 needs to be translated

**Comment by wilsonge** [comment]
@infograf768 do we have any information from Mig about how this is going?

**Comment by infograf768** [comment]
@wilsonge
You folks have to contact him to organise this.
Some of the new files should be posted in the Translation Teams Forum and then merged.
What is not so simple is that some must be provided by TTs (as the ones available in Tiny specific page are not good enough) and others have to be used as they are (after some name modifications to fit our needs) from the Tiny site.
As far as I know, nothing was done yet.

If this is the only aspect which prevents you from merging this PR, no worry. 
Updated translation files can be gathered for next release.
Someone though should test this PR

**Comment by wilsonge** [comment]
@dgt41 Can we upgrade this to tiny 4.2.5 please. I'm talking with mig at the moment for the translation aspect

**Comment by dgt41** [comment]
@wilsonge 4.2.5 done!

**Comment by dgt41** [comment]
@wilsonge @infograf768 about the translation: the new strings are for the image tools which Joomla is not using right now, and I will say neither any of the users should use it at the time, as they will end up saving the images in their database. To cut it short, translation of those strings is welcome but not mandatory at the moment

**Comment by infograf768** [comment]
> translation of those strings is welcome but not mandatory at the moment

Indeed.

**Comment by infograf768** [comment]
The Tiny version in the xml needs updating.

Other than that, it looks like working BUT something important has changed for the language files (will inform TTs and Thomas):

The first line of each xx.js or xx-XX.js has to contain the same tag as the name of the file (as they exist already in core).

Example for French `fr.js`
It has as first line 
`tinymce.addI18n('fr_FR',{`

it should be
`tinymce.addI18n('fr',{`

For Brazilian `pt-BR.js`
it has `tinymce.addI18n('pt_BR',{`
it should now be
`tinymce.addI18n('pt-BR',{`

**Comment by infograf768** [comment]
Ok for me. One more tester.

**Comment by wilsonge** [comment]
It looks like all the 3.x compatibility plugins have been removed :/

File: media/editors/tinymce/plugins/compat3x/tiny_mce_popup.js [hunk 1]
```diff
@@ -19,47 +18,55 @@ var tinymce = null, tinyMCEPopup, tinyMCE;
...
+		settings = self.editor.settings;
 
 		// Setup popup CSS path(s)
 		if (settings.popup_css !== false) {
 			if (settings.popup_css) {
-				settings.popup_css = t.documentBaseURI.toAbsolute(settings.popup_css);
+				settings.popup_css = self.editor.documentBaseURI.toAbsolute(settings.popup_css);
 			} else {
-				settings.popup_css = t.baseURI.toAbsolute("themes/" + settings.theme + "/skins/" + settings.skin + "/dialog.css");
+				settings.popup_css = self.editor.baseURI.toAbsolute("plugins/compat3x/css/dialog.css");
```
Comment (dgt41) [inline media/editors/tinymce/plugins/compat3x/tiny_mce_popup.js]: This line explains why there is a css and an omg folder in the compat3x folder. Also the new skin dropped most of the icons...

**Comment by infograf768** [comment]
Shall we not delete all compat3x folder and add these files in com_admin/script.php as anyway they are never used in J3.x ?

**Comment by dgt41** [comment]
This will simplify the update process for tinyMCE, but it’s PLT’s call

**Comment by slibbe** [comment]
I have tested this item :white_check_mark: successfully on 6b66a1a48c1d9e9413654f7a5aa5c8d5341704de<br /><br />I have tested this and it seems to work ok. However, I don't know what the bugs with the older version were or what other attention points there may be.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7667">issues.joomla.org/joomla-cms/7667</a>.</sub>

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 6b66a1a48c1d9e9413654f7a5aa5c8d5341704de<br /><br />OK here<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7667">issues.joomla.org/joomla-cms/7667</a>.</sub>

**Comment by zero-24** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7667">issues.joomla.org/joomla-cms/7667</a>.</sub>

**Comment by infograf768** [comment]
Please inform TTs when this is merged as quite a few tiny langs have to be updated

**Comment by Bakual** [comment]
@infograf768 I've updated the language files with https://github.com/joomla/joomla-cms/commit/458b617dafa8d8b2d2a72ef71bb1b7cb154d6fdd using the packs provided by the translators and also fixed the tags you mentioned in the thread. Can you have a look over it if it's correct?

**Comment by infograf768** [comment]
Already on it ;)

**Comment by infograf768** [comment]
Looks OK as far as I could see.

**Pull request merged**